org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+            Debug.WriteLine( string.Format( "Comparing {0} Top-Level Domains.", tlds.Length ) );
+
+            // Get HS API Token
+            string attrKey = dataMap.GetString("AttributeKey");
+            string attrValue = GlobalAttributesCache.Get().GetValue(attrKey);
+            key = Encryption.DecryptString(attrValue);
```

Comment:
If `key` is blank, the job can't function properly. Can we add some validation here that throws an exception and stops the job from running if `key` is blank?